lib/oauth/factory.go

```diff
@@ -389,19 +397,60 @@ func WithFlags(fl *Flags) Modifier {
 			fl.TokenVerifyingKey = (*verify.ToBytes())[:]
 		}
 
-		mods = append(mods, WithAuthTime(fl.AuthTime), WithSecrets(fl.OauthSecretID, fl.OauthSecretKey), WithSigningExtractorFlags(fl.SigningExtractorFlags))
+		mods = append(mods,
+			WithAuthTime(fl.AuthTime),
+			WithSecrets(fl.OauthSecretID, fl.OauthSecretKey),
+			WithSigningExtractorFlags(fl.SigningExtractorFlags),
+			WithAdditionalFlow(fl.AdditionalOAuth),
+		)
 		return Modifiers(mods).Apply(o)
 	}
 }
 
+func WithAdditionalFlow(fileContent []byte) Modifier {
```

Comment:
this function can be turned into an extra flag in `oauth/factory.go` to create a provider from a single json file? It would be a good improvement?

But not needed strictly in this PR now?

Reply:
It's already used this way, and registered as a flag <.<

Reply:
WithAdditionalFlows should be renamed as WithJsonConfig?